13_import.sql

```diff
@@ -0,0 +1,536 @@
+    ma_material,
+    ma_dimension1,
+    ma_dimension2,
+    ws_type,
+    ma_function,
+    ss_function,
+    remark,
+    wn_bottom_level
+    )
+    VALUES
+    (
+    NEW.obj_id,
+    NEW.identifier,
+    multi_situation_geometry,
+    NEW.co_shape,
+    NEW.co_diameter,
+    NEW.co_material,
+    NEW.co_positional_accuracy,
+    NEW.co_level,
+    NEW._depth,
+    NEW._channel_usage_current,
+    NEW.ma_material,
+    NEW.ma_dimension1,
+    NEW.ma_dimension2,
+    NEW.ws_type,
+    NEW.ma_function,
+    NEW.ss_function,
+    NEW.remark,
+    NEW.wn_bottom_level
+    );
+    RAISE NOTICE 'Inserted row in qgep_od.vw_qgep_wastewater_structure';
+  END IF;
+
+  -- photo1 insert
+  IF (NEW.photo1 IS NOT NULL) THEN
+    INSERT INTO qgep_od.file
+    (
+      object,
+      identifier
+    )
+    VALUES
+    ( 
+      NEW.obj_id,
+      NEW.photo1
+    );
+    RAISE NOTICE 'Inserted row in qgep_od.file';
+  END IF;
+
+  -- photo2 insert
+  IF (NEW.photo2 IS NOT NULL) THEN
+    INSERT INTO qgep_od.file
+    (
+      object,
+      identifier
+    )
+    VALUES
+    ( 
+      NEW.obj_id,
+      NEW.photo2
+    );
+    RAISE NOTICE 'Inserted row in qgep_od.file';
+  END IF;
+
+  -- set structure okay
+  UPDATE qgep_import.manhole_quarantine
+  SET structure_okay = true
+  WHERE quarantine_serial = NEW.quarantine_serial;
+  RETURN NEW;
+
+  -- catch
+  EXCEPTION WHEN OTHERS THEN
+    RAISE NOTICE 'EXCEPTION: %', SQLERRM;
+    RETURN NEW;
+END; $BODY$
+LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS after_update_try_structure_update ON qgep_import.manhole_quarantine;
+
+CREATE TRIGGER after_update_try_structure_update
+  AFTER UPDATE
+  ON qgep_import.manhole_quarantine
+  FOR EACH ROW
+  WHEN ( ( NEW.structure_okay IS NOT TRUE ) 
+  AND NOT( OLD.inlet_okay IS NOT TRUE AND NEW.inlet_okay IS TRUE )
+  AND NOT( OLD.outlet_okay IS NOT TRUE AND NEW.outlet_okay IS TRUE ) )
+  EXECUTE PROCEDURE qgep_import.manhole_quarantine_try_structure_update();
+
+DROP TRIGGER IF EXISTS after_insert_try_structure_update ON qgep_import.manhole_quarantine;
+
+CREATE TRIGGER after_insert_try_structure_update
+  AFTER INSERT
+  ON qgep_import.manhole_quarantine
+  FOR EACH ROW
+  EXECUTE PROCEDURE qgep_import.manhole_quarantine_try_structure_update();
+
+-- Some information:
+-- 1. new lets 0 - old lets 0 -> do nothing
+-- 2. new lets 0 - old lets 1 -> delete let
+-- 3. new lets 0 - old lets n -> delete lets
+-- 4. new lets 1 - old lets 0 -> create let (future)
+-- 5. new lets 1 - old lets 1 -> update let
+-- 6. new lets 1 - old lets n -> manual update needed
+-- 7. new lets n - old lets 0 -> create lets (future)
+-- 8. new lets n - old lets 1 -> manual update needed
+-- 9. new lets n - old lets n -> manual update needed
+
+CREATE OR REPLACE FUNCTION qgep_import.manhole_quarantine_try_let_update() RETURNS trigger AS $BODY$
+  DECLARE 
+    is_inlet boolean;
+    let_kind text;
+    new_lets integer;
+    old_lets integer;
+BEGIN
+  is_inlet := TG_ARGV[0];
```

Comment:
I might be a bit picky, but let's pass directly the string as argument, that would be clearer in the trigger code, i.e. 
`EXECUTE PROCEDURE qgep_import.manhole_quarantine_try_let_update( 'inlet' );` 
instead of 
`EXECUTE PROCEDURE qgep_import.manhole_quarantine_try_let_update( TRUE );`

Reply:
Done